poc_iot_injector/src/server.rs

```diff
@@ -151,9 +153,18 @@ async fn submit_txns(
     let mut stream =
         store.source_unordered(LOADER_WORKERS, stream::iter(file_list).map(Ok).boxed());
 
+    let max_txns_per_min =
+        NonZeroU32::new(settings.max_txns_per_min).unwrap_or(nonzero!(MAX_TXNS_PER_MIN));
+    let bucket = Arc::new(RateLimiter::direct(Quota::per_minute(max_txns_per_min)));
+
+    // To ensure that multiple tasks don't wake up at the exact same time
+    let jitter = Jitter::up_to(StdDuration::from_secs(5));
+
     let mut set = JoinSet::new();
 
     while let Some(msg) = stream.next().await {
+        let bucket = Arc::clone(&bucket);
```

Comment:
since bucket is already inside an Arc can't you just `bucket.clone()` here? and do you need the `&` since an Arc is one of of the standard smart pointers?

Reply:
Yeah, updated [cb33e0c](https://github.com/helium/oracles/pull/324/commits/cb33e0ce8a27c101543b6b34e753f431d1291d84)